source/core_json.c

```diff
@@ -1233,21 +1301,19 @@ static bool_ nextKeyValuePair( const char * buf,
  *
  * Iterate over the key-value pairs of an object, looking for a matching key.
  *
- * @return #JSONSuccess if the queryKey is found and the value output;
- * #JSONIllegalDocument if the buffer contents are NOT valid JSON;
- * #JSONMaxDepthExceeded if object and array nesting exceeds a threshold;
- * #JSONNotFound if the queryKey is NOT found.
+ * @return true if the queryKey is found and the value output;
+ * false otherwise.
  *
  * @note Parsing stops upon finding a match.
  */
-static JSONStatus_t search( char * buf,
-                            size_t max,
-                            const char * queryKey,
-                            size_t queryKeyLength,
-                            char ** outValue,
-                            size_t * outValueLength )
+static bool_ objectSearch( char * buf,
```

Comment:
Should the `buf` parameter be of `const char *` type?

Reply:
No, because of the aliasing I mentioned above.